week-5/Homework/mandatory/2-exercises/exercises.js

```diff
@@ -1,64 +1,119 @@
+ you need to implement functions' body.
+ */
+
+/*
+ Optional question:
+
+ Notice how this JavaScript file linked to the index.html file.
+ What do you think when this JavaScript file is being executed?
+
+ When you finished this exercise try to move the script tag around the Div tag with the id "content",
+ refresh the page and observe what happens. (This should answer the question above.)
+ 
+ What do you think why this changes shouldn't work?
+ */
+// LEAVE YOUR ANSWER HERE (THIS IS OPTIONAL)
+
 /**
+ * This function recieves a lists of people. Each object should contain the name and the occupation of a person.
+ * Look for usage of the function in the code and see what variable is passed into it as an argument.
  *
- * For each of the names in the array passed into this function
- *
- * - Add a <h1> tag to the website containing the name of the person
- * - Add a <h2> tag to the website containing the job of the person
+ * Create and insert the below HTML elements to the DOM for each of Objects in the Array as follows:
+ *  1. Add a <h1> tag to the website containing the name of the person
+ *  2. Add a <h2> tag to the website containing the job of the person
  *
- * All of your HTML should go inside the Div tag with the id "content".
+ * All of your HTML elements should go inside the Div tag with the id "content".
  *
+ * An example "content" div should look like:
  * <div id="content">
- *      <h1>{Name Here}</h1>
- *      <h2>{Job Here}</h2>
- *      .....
+ *      <h1>Mario</h1>
+ *      <h2>Plumber</h2>
+ *      <h1>Luigi</h1>
+ *      <h2>Plumber</h2>
  * </div>
  */
-function exerciseOne(arrayOfPeople) {
+function insertPeopleData(arrayOfPeople) {
   let content = document.querySelector("#content");
+  for (let i = 0; i < arrayOfPeople.length; i++) {
+    var h1El = document.createElement("h1");
+    h1El.textContent = arrayOfPeople[i].name;
+    content.appendChild(h1El);
+    var h2El = document.createElement("h2");
+    h2El.textContent = arrayOfPeople[i].job;
+    content.appendChild(h2El);
+  }
 }
 
 /**
  *
- * Create a list of shopping items. You should use an unordered list.
+ * Create a list of shopping items using an unordered HTML list.
+ * The input of this function is a simple Array of Strings, look for the concrete example in the code.
  *
- * All of your HTML should go inside the Div tag with the id "content".
+ * All of your HTML elements should go inside the Div tag with the id "content".
  *
+ * Hint for type of lists in HTML: https://www.w3schools.com/html/html_lists.asp
  */
-function exerciseTwo(shopping) {
-  //Write your code in here
+function insertShoppingList(shoppingList) {
+  let content = document.querySelector("#content");
+  let listToBuy = document.createElement("ul");
+  for (let i = 0; i < shoppingList.length; i++) {
+    let list = document.createElement("li");
+    list.textContent = shoppingList[i];
+    listToBuy.appendChild(list);
+    content.appendChild(listToBuy);
```

Comment:
Well done! I can see you used a for loop but remember that higher order functions such as `forEach` are your friend here, try to use them when possible🙂 Also appending `ul` element (on line 67) to the content should be outside the loop